### PR TITLE
LFTR chemistry fixes + samarium (III) chloride chemical formula fix

### PR DESCRIPTION
**Pull request by hallucinogender**
Fixed formula for "Samarium(III)-Chloride" material to actually be samarium (III) chloride rather than samarium carbide iodide. This is independent of the rest.

### LFTR Chemistry Fixes (First Pass)
The following list of changes are the first stage of fixing the chemistry associated with the Liquid Fluoride Thorium Reactor (LFTR) to be both internally and externally consistent. This fixes numerous duplication bugs, transmutation bugs, and other erroneous chemistry. Doing this has inevitably changed the way the reactor is balanced, and further adjustments are my next priority to ensure that the reactor balance is still suitable. The radioisotope fuels for the LFTR are unchanged by any of my edits, but other materials (such as lithium-7, beryllium, zirconium, and fluorine especially) are balanced differently in how the reactor uses them compared to before. This is necessary in order to remove the numerous unavoidable duplication bugs and other variously broken chemistry involved in preparing the molten salts for the reactor.

Note: for affected materials with both an item and fluid form, I generally use the fluid rule (https://wiki.gtnewhorizons.com/wiki/Chemical_Balance) because most chemistry relating to the LFTR happens in fluid form. I am treating these materials as though the fluid is the default. The degree of inconsistency in the original recipes meant that there wasn't a clear preference, so I've chosen to favour this approach here since it worked out better in most cases while I was experimenting with possible fixes. The only material that this isn't applicable to is lithium hydroxide, since the fluid form of that is independent of everything else.

Additionally, note that "Industrial Strength Hydrofluoric Acid" follows the same convention in my newly-corrected chemistry as it uses elsewhere: industrial strength HF is considered to be twice as dense as regular hydrofluoric acid, therefore 1000 liters of industrial strength is equivalent to 2000 liters of regular hydrofluoric acid, and thus every 1000 liters of industrial strength hydrofluoric acid contains 2000 liters of hydrogen and 2000 liters of fluorine. This unique convention applies very often in my edited chemistry, where I have made it consistent with the way industrial strength hydrofluoric acid is both manufactured (in the chemical plant, particularly) and utilized elsewhere.

1. Fixed chemical formula for "Li2CO3 + Ca(OH)2 Compound Dust". Chemical formula is missing a set of brackets, and this typo appears to have led to other errors with recipes involving the material.
2. Changed item:fluid ratio from 1:144 to 1:1000 for most materials relating to the LFTR with both item and fluid form. The default is 1:144, but some of the erroneous recipes treat it as a 1:144 ratio while others treat it as a 1:1000 ratio with very little consistency at all. I have chosen to change the ratio like this because doing so requires less tinkering with the nuclear fuel reprocessing recipes (sodium fluoride, beryllium fluoride, and lithium fluoride involved in spent fuel reprocessing all influence this decision). Recipes changed for this include fluid extractor recipes from solid to fluid form of various materials, dehydrator recipes from fluid to solid form of various materials, and one fluid heater recipe for melting sodium fluoride. This change doesn't apply to lithium hydroxide because the fluid form of that material is independent of everything else.
3. Fixed amount of chlorine required for zirconium pellet recipe in autoclave. Each ZrCl4 item is now composed of 1 zirconium dust and 4000 liters of chlorine. This (plus another later change relating to zirconium chloride) fixes a chlorine duplication issue when manufacturing zirconium tetrafluoride.
4. Fixed both dehydrator recipes for zirconium tetrafluoride dust to consume correct amount of fluorine and remove chlorine duplication issue.
5. Adjusted EBF recipe for lithium tetrafluoroberyllate fuel compound dust to obey my new chemistry rules. This change is part of making a single unit of lithium tetrafluoroberyllate fuel compound dust equate to exactly one unit of its entire chemical formula (due to favouring the fluid rule for internal consistency with chemistry here).
6. Fixed production of lithium fluoride dust" from lithium carbonate dust in chemical bath to remove lithium-7 duplication and to use the correct amount of industrial strength hydrofluoric acid.
7. Fixed production of lithium fluoride dust from lithium hydroxide dust in chemical reactor to remove lithium-7 duplication and to use the correct amount of industrial strength hydrofluoric acid, and to output the correct amount of water byproduct.
8. Fixed production of "Li2CO3 + Ca(OH)2 Compound Dust" in chemical reactor to use the correct proportions of the reagents and to output the correct amount of the product.
9. Fixed dehydrator recipe for "Li2CO3 + Ca(OH)2 Compound Dust" to consume the correct amount of the reagent and to output the correct amounts of the products. Also doubled recipe duration to preserve overall duration of this process when combined with the previous process.
10. Fixed production of lithium hydroxide fluid in chemical reactor. Now consumes correct quantities of water and oxygen, and outputs correct amount of fluid to fix deletion of majority of lithium-7 input.
11. Fixed dehydrator recipe for beryllium fluoride to use standardized chemistry balance rules and remove duplication of industrial strength hydrofluoric acid and ammonia during production of beryllium fluoride.
12. Fixed dehydrator recipe for ammonium tetrafluoroberyllate to use standardized chemistry balance rules and remove duplication of industrial strength hydrofluoric acid and ammonia during production of beryllium fluoride.
13. Fixed chemical reactor recipe for beryllium hydroxide to use standardized chemistry balance rules.
14. Fixed both chemical reactor recipes for ammonium bifluoride to use standardized chemistry balance rules and remove duplication of industrial strength hydrofluoric acid and ammonia during production of beryllium fluoride. I also skip the production of the ammonium ion as a separate fluid; energy consumption and overall time have been merged into this recipe; this was the most sensible way to ensure that the correct amounts of both fluorine and hydrogen are used.
15. Fixed both chemical reactor recipes for sodium fluoride to use standardized chemistry balance rules and obey my aforementioned conventions for repairing the various chemistry balance errors here. This does affect nuclear fuel reprocessing balance somewhat by requiring less sodium than previously, but means that there is neither fluorine duplication nor substantial fluorine deletion in the stages that require sodium fluoride.
16. Fixed mixer recipe for uranium tetrafluoride to use correct amount of industrial strength hydrofluoric acid (twice as much as previous) so that uranium tetrafluoride fluid now obeys fluid rule in chemistry balance.

Overall, these changes have repaired numerous chemistry errors (including numerous duplication bugs while manufacturing fluorides of lithium-7, beryllium, and zirconium) associated with the LFTR. The reactor fuel cycle, including spent fuel reprocessing, remains unchanged, which means that the balance is the same for nuclear isotopes. The reactor does now require less lithium and zirconium to operate, but needs either the same amount of fluorine or possibly more fluorine (though this is difficult to analyse). As far as I can tell, there is no more broken chemistry and there are no more duplication bugs. However, while there is no fluorine duplication _at present_, I will be taking a closer look at the fluorine cycle within the LFTR and its associated fuel processing recipes because fluorine will likely become net positive if the currently-useless fluoride outputs of the sparge tower ever gain uses (which is a change I'd like to make in future so that the reactor feels less unfinished, by having it no longer output multiple distinct useless byproducts).

To summarize in full, these changes have fixed things that are currently very badly broken (including the fact that it is currently impossible to even operate the LFTR without duplicating calcium, lithium-7, industrial strength hydrofluoric acid, chlorine, hydrogen, ammonia, and possibly other materials as well). They don't break anything as they currently stand, and they don't rebalance the radioisotope fuels of the LFTR (though they do necessarily rebalance some of its other inputs). Further adjustments will be needed in future when I do more work on the LFTR, but these changes here constitute the bare minimum of repairs necessary to fix currently extant (and rather severe) chemistry bugs surrounding the LFTR.

### CODE REVIEW

**Comment by chochem** [comment]
> Fixed chemical formula for "Li2CO3 + Ca(OH)2 Compound Dust". Chemical formula is missing a set of brackets, and this typo appears to have led to other errors with recipes involving the material.

if its actually a compound dust then its missing 2 sets of brackets, no? right now: Li2CO3CaOH2 but should be (Li2CO3)(Ca(OH)2)

**Comment by chochem** [comment]
> Changed item:fluid ratio from 1:144 to 1:1000 for most materials relating to the LFTR with both item and fluid form. The default is 1:144, but some of the erroneous recipes treat it as a 1:144 ratio while others treat it as a 1:1000 ratio with very little consistency at all. I have chosen to change the ratio like this because doing so requires less tinkering with the nuclear fuel reprocessing recipes (sodium fluoride, beryllium fluoride, and lithium fluoride involved in spent fuel reprocessing all influence this decision). Recipes changed for this include fluid extractor recipes from solid to fluid form of various materials, dehydrator recipes from fluid to solid form of various materials, and one fluid heater recipe for melting sodium fluoride. This change doesn't apply to lithium hydroxide because the fluid form of that material is independent of everything else.

please follow: molten stuff = 144, everything else = 1000
or is there any reason not to follow that?

e.g. these ones are molten and should stay at 144 
image

**Comment by chochem** [comment]
that probably then also affects:
> for affected materials with both an item and fluid form, I generally use the fluid rule

the rule is clear: molten = use solid rule. otherwise fluid rule

**Comment by hallucinogender** [comment]
I've made a second round of fixes. Everything is now balanced based around the original dust:fluid ratios for things, and I've gone through and checked in-game that all of the recipes work and are accurate. The chemistry is, again, fixed and correct, but this time the fluid to solid ratio stays the same.

Additionally, uranium tetrafluoride has been changed for futureproofing, since there _is_ a dust form (though one that is currently unobtainable). Uranium tetrafluoride now also obeys the solid rule for chemistry balancing (i.e. 5 UF4 dust = 1 uranium dust + 4000 fluorine), so that if the dust form is ever utilized, existing recipes won't cause problems.

File: src/main/java/gtPlusPlus/xmod/gregtech/loaders/recipe/RecipeLoader_Nuclear.java [hunk 1]
```diff
@@ -198,27 +198,27 @@ private static void chemicalReactorRecipes() {
             .getItemStackOfAmountFromOreDictNoBroken("cellHydrofluoricAcid_GT5U", 2);
 
         GT_Values.RA.addChemicalRecipe(
-            ItemUtils.getItemStackOfAmountFromOreDict("dustLithiumCarbonate", 5), // Input
+            ItemUtils.getItemStackOfAmountFromOreDict("dustLithiumCarbonate", 6), // Input
             ItemUtils.getItemStackOfAmountFromOreDict("dustCalciumHydroxide", 5), // Input
             null, // Fluid Input
             null, // Fluid Output
-            ItemUtils.getItemStackOfAmountFromOreDict("dustLi2CO3CaOH2", 10),
+            ItemUtils.getItemStackOfAmountFromOreDict("dustLi2CO3CaOH2", 11),
```
Comment (chochem) [inline src/main/java/gtPlusPlus/xmod/gregtech/loaders/recipe/RecipeLoader_Nuclear.java:205]: this recipe breaks chem balance.

File: src/main/java/gtPlusPlus/xmod/gregtech/loaders/recipe/RecipeLoader_Nuclear.java [hunk 2]
```diff
@@ -350,13 +351,13 @@ private static void dehydratorRecipes() {
         // 2 LiOH + CaCO3
         CORE.RA.addDehydratorRecipe(
             new ItemStack[] { CI.getNumberedAdvancedCircuit(20),
-                ItemUtils.getItemStackOfAmountFromOreDict("dustLi2CO3CaOH2", 5) }, // Item
+                ItemUtils.getItemStackOfAmountFromOreDict("dustLi2CO3CaOH2", 11) }, // Item
```
Comment (chochem) [inline src/main/java/gtPlusPlus/xmod/gregtech/loaders/recipe/RecipeLoader_Nuclear.java:354]: this is very likely wrong too then

**Comment by hallucinogender** [comment]
My most recent changes adjust the chemical formula of hydrated lime (calcium hydroxide) to be consistent with every recipe that involves hydrated lime, and adjust the name and formula of the lithium carbonate + hydrated lime compound to obey the chemistry balance rules that are actually being used.

I ran spotlessApply but it didn't change anything.

File: src/main/java/com/elisis/gtnhlanth/common/register/WerkstoffMaterialPool.java [hunk 3]
```diff
@@ -1106,7 +1106,7 @@ public class WerkstoffMaterialPool implements Runnable {
     public static final Werkstoff SamariumChloride = new Werkstoff(
         new short[] { 248, 248, 120 },
         "Samarium(III)-Chloride",
-        subscriptNumbers("SmCI3"),
+        subscriptNumbers("SmCl3"),
```
Comment (chochem) [inline src/main/java/com/elisis/gtnhlanth/common/register/WerkstoffMaterialPool.java:1109]: this probably should have a ? as both the previous and the following step in the processing have one, To indicate its not pure and has some unknown other stuff in there with unknown quantity. So ?SmCl3

**Review by chochem — COMMENTED** [review]
Outside of that small point it looks good now. thanks a lot!

**Review by chochem — APPROVED** [review]

**Comment by hallucinogender** [comment]
Apologies for leaving this long enough that someone else has (apparently) had to make that final requested change. I didn't have my medication for several weeks so was unable to focus on learning how to manage merge conflicts.

**Comment by chochem** [comment]
no problem, was a very small thing anyway.
